diff --git a/dependency_support/requirements.bzl b/dependency_support/requirements.bzl
index b3361e6..6b1c67d 100644
--- a/dependency_support/requirements.bzl
+++ b/dependency_support/requirements.bzl
@@ -1,4 +1,4 @@
-_config = {"download_only": False, "enable_implicit_namespace_pkgs": False, "environment": {}, "envsubst": [], "extra_pip_args": [], "isolated": True, "pip_data_exclude": [], "python_interpreter": "python3", "python_interpreter_target": "@python39_host//:python", "quiet": True, "repo": "rules_hdl_pip_deps_vendored", "repo_prefix": "rules_hdl_pip_deps_vendored_", "timeout": 600}
+_config = {"download_only": False, "enable_implicit_namespace_pkgs": False, "environment": {}, "envsubst": [], "extra_pip_args": ["--no-cache-dir"], "isolated": True, "pip_data_exclude": [], "python_interpreter": "python3", "python_interpreter_target": "@python39_host//:python", "quiet": True, "repo": "rules_hdl_pip_deps_vendored", "repo_prefix": "rules_hdl_pip_deps_vendored_", "timeout": 600}
 _annotations = {}
 
 def requirement(name):
