diff --git python_configure.bzl python_configure.bzl
index 1f5bffa..4225255 100644
--- python_configure.bzl
+++ python_configure.bzl
@@ -252,8 +252,8 @@ def _get_python_include(repository_ctx, python_bin):
             python_bin,
             "-c",
             "from __future__ import print_function;" +
-            "from distutils import sysconfig;" +
-            "print(sysconfig.get_python_inc())",
+            "import sysconfig; import os; " +
+            "print(os.path.dirname(sysconfig.get_config_h_filename()))",
         ],
         error_msg = "Problem getting python include path.",
         error_details = ("Is the Python binary path set up right? " +
