From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Rigge <rigge@google.com>
Date: Thu, 22 Feb 2024 13:32:05 -0800
Subject: Remove duplicate logtostderr flag.


diff --git a/ortools/base/logging.cc b/ortools/base/logging.cc
index d04eb6e037..c88adac71c 100644
--- a/ortools/base/logging.cc
+++ b/ortools/base/logging.cc
@@ -20,8 +20,6 @@
 #include "absl/log/globals.h"
 #include "absl/log/initialize.h"
 
-ABSL_FLAG(bool, logtostderr, false, "no op compatibility flag");
-
 namespace operations_research {
 
 namespace {
-- 
2.44.0.rc0.258.g7320e95886-goog

