diff --git ortools/linear_solver/BUILD.bazel ortools/linear_solver/BUILD.bazel
index 85e3bdfd88..b6f00dc01a 100644
--- ortools/linear_solver/BUILD.bazel
+++ ortools/linear_solver/BUILD.bazel
@@ -96,9 +96,6 @@ cc_library(
     }) + select({
         ":with_xpress": ["xpress_interface.cc"],
         "//conditions:default": [],
-    }) + select({
-        ":with_glpk": ["glpk_interface.cc"],
-        "//conditions:default": [],
     }),
     hdrs = [
         "glop_interface.cc",
@@ -113,10 +110,7 @@ cc_library(
         "-DUSE_PDLP",
         "-DUSE_SCIP",
         "-DUSE_HIGHS",
-    ] + select({
-        ":with_glpk": ["-DUSE_GLPK"],
-        "//conditions:default": [],
-    }),
+    ],
     deps = [
         ":linear_solver_cc_proto",
         ":scip_with_glop",
@@ -152,11 +146,7 @@ cc_library(
         "//ortools/sat:lp_utils",
         "//ortools/util:fp_utils",
         "//ortools/util:lazy_mutable_copy",
-    ] + select({
-        ":with_glpk": ["@glpk//:glpk", 
-        "//ortools/glpk:glpk_env_deleter"],
-        "//conditions:default": [],
-    }),
+    ],
 )
 
 cc_library(
