diff --git ortools/linear_solver/BUILD.bazel ortools/linear_solver/BUILD.bazel
index 9c9a1d0a6c..a5cecf9d97 100644
--- ortools/linear_solver/BUILD.bazel
+++ ortools/linear_solver/BUILD.bazel
@@ -58,10 +58,7 @@ cc_library(
         "linear_expr.cc",
         "linear_solver_callback.cc",
         "linear_solver.cc",
-        "lpi_glop.cpp",
         "model_validator.cc",
-        "pdlp_interface.cc",
-        "pdlp_proto_solver.cc",
         "sat_interface.cc",
         "sat_proto_solver.cc",
         "sat_solver_utils.cc",
@@ -100,12 +97,9 @@ cc_library(
         "scip_proto_solver.h",
     ],
     copts = [
-        "-DUSE_PDLP",
-        "-DUSE_SCIP",
     ],
     deps = [
         ":linear_solver_cc_proto",
-        ":scip_with_glop",
         ":model_exporter",
         "@com_google_absl//absl/status",
         "@com_google_absl//absl/status:statusor",
@@ -124,11 +118,7 @@ cc_library(
         "//ortools/bop:integral_solver",
         "//ortools/glop:lp_solver",
         "//ortools/glop:parameters_cc_proto",
-        "//ortools/gscip:legacy_scip_params",
         "//ortools/gurobi:environment",
-        "//ortools/pdlp:primal_dual_hybrid_gradient",
-        "//ortools/pdlp:solve_log_cc_proto",
-        "//ortools/pdlp:solvers_cc_proto",
         "//ortools/port:file",
         "//ortools/port:proto_utils",
         "//ortools/sat:cp_model_cc_proto",
@@ -142,21 +132,6 @@ cc_library(
     }),
 )
 
-copy_file(
-    name = "lpi_glop",
-    src = "@scip//:src/lpi/lpi_glop.cpp",
-    out = "lpi_glop.cpp",
-)
-
-cc_library(
-    name = "scip_with_glop",
-    srcs = ["lpi_glop.cpp"],
-    deps = [
-        "//ortools/glop:lp_solver",
-        "@scip//:libscip",
-    ],
-)
-
 cc_library(
     name = "scip_helper_macros",
     hdrs = ["scip_helper_macros.h"],
