diff --git a/common/lsp/BUILD b/common/lsp/BUILD
index 8273dc26..cf0b71ae 100644
--- a/common/lsp/BUILD
+++ b/common/lsp/BUILD
@@ -11,7 +11,7 @@ load("//bazel:sh_test_with_runfiles_lib.bzl", "sh_test_with_runfiles_lib")
 package(
     default_applicable_licenses = ["//:license"],
     default_visibility = [
-        "//verilog/tools/ls:__subpackages__",
+        "//visibility:public",
     ],
     features = ["layering_check"],
 )
