--- boost.BUILD	2023-07-22 21:40:32.570049258 -0700
+++ boost.BUILD	2023-07-23 07:33:08.760174104 -0700
@@ -1901,18 +1901,6 @@
         "//conditions:default": [],
     }),
     exclude_src = ["libs/stacktrace/src/*.cpp"],
-    linkopts = select({
-        ":linux_ppc": [
-            "-lbacktrace -ldl",
-        ],
-        ":linux_x86_64": [
-            "-lbacktrace -ldl",
-        ],
-        ":linux_aarch64": [
-            "-lbacktrace -ldl",
-        ],
-        "//conditions:default": [],
-    }),
     deps = [
         ":array",
         ":config",
@@ -1922,6 +1910,7 @@
         ":predef",
         ":static_assert",
         ":type_traits",
+        "@com_github_libbacktrace//:libbacktrace",
     ],
 )
 
