diff --git a/src/lib/bitblast/aig/aig_manager.cpp b/src/lib/bitblast/aig/aig_manager.cpp
--- a/src/lib/bitblast/aig/aig_manager.cpp
+++ b/src/lib/bitblast/aig/aig_manager.cpp
@@ -129,7 +129,10 @@ AigManager::AigManager()
   assert(d_false.get_id() == -AigNode::s_true_id);
 }
 
-AigManager::~AigManager() {}
+AigManager::~AigManager() {
+  d_true = AigNode();
+  d_false = AigNode();
+}
 
 const AigManager::Statistics&
 AigManager::statistics() const
