From 160a546da6ce0a76b160b57e6350f610ccf8b808 Mon Sep 17 00:00:00 2001
From: Paul Rigge <rigge@google.com>
Date: Mon, 22 Jul 2024 20:59:04 -0700
Subject: [PATCH] Add absl:status to deps.

---
 src/core/BUILD | 1 +
 1 file changed, 1 insertion(+)

diff --git src/core/BUILD src/core/BUILD
index e76dc300ac..0688bb9dcc 100644
--- src/core/BUILD
+++ src/core/BUILD
@@ -2563,6 +2563,7 @@ grpc_cc_library(
     external_deps = [
         "absl/container:flat_hash_map",
         "absl/log:check",
+        "absl/status",
         "absl/strings",
         "absl/strings:str_format",
     ],
--
2.45.2

